### DISCUSSION TITLE
Discussion thread for RFC 1027 - Simplfying path resolution

### DISCUSSION THREAD

**Discussion by msullivan** [geldata/gel#7824] [Ideas]
I'm proposing a backwards incompatible change to EdgeQL that should substantially simplify the semantics of the language without affecting most queries: [RFC 1027 - Simplifying path resolution](https://github.com/edgedb/rfcs/blob/master/text/1027-no-factoring.rst)

I'm interested in community feedback about:

- The changes themselves
- Migration planning

**Comment by jackfischer** [comment]
Personally I never invested the time to grok what was going on here because I never got a motivation for it from a querying perspective, just assumed it was somehow an logical consequence of other parts of the design and that it was unlikely to matter in practice. From that perspective this is not a big deal. #5575 was a disaster for us fwtw. If the current behavior also makes it harder to work on edgedb itself which it sounded like it does, that also has negative higher order effects on what users can get out of edgedb just due to how fast edgedb can ship with finite resources.

The migration plan and concept of supporting both actually looks quite generous. I strongly suspect most people will get _no_ warnings. Why not make this a breaking change for the major version release and put any saved effort towards 6.x in general or other issues? That may well be net more developer friendly and adoption friendly.
